staging/src/k8s.io/apiserver/pkg/endpoints/filters/authentication.go

```diff
@@ -170,3 +174,19 @@ func compressUsername(username string) string {
 		return "other"
 	}
 }
+
+func authenticatorIDFromResponse(response *authenticator.Response) string {
+	if response == nil || response.AuthenticatorName == "" {
+		return unknownAuthenticator
+	}
+	return response.AuthenticatorName
+}
+
+func authenticatorIDFromError(err error) string {
+	var authErr *authenticator.Error
+	if ok := errors.As(err, &authErr); !ok {
```

Comment:
Did you mean to pass in a `**authenticator.Error`?

Reply:
Note sure I follow this.
Added unit tests, PTAL.